### PR TITLE
Add support to provide JMS properties when creating the connection

### PR DESCRIPTION
**Pull request by AnuGayan**

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java [hunk 1]
```diff
@@ -99,8 +107,8 @@
                                 + "password='1920',"
                                 + "username='mqm',"
                                 + "@map(type='text'))"
-                                + "define stream SweetProductionStream(name string, amount double);",
-                        description = "This example shows how to connect to an IBM MQ queue and send messages."),
+                                + "define stream SweetProductionStream(name string, amount double);"
```
Comment (minudika) [inline component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java:113]: shall we add the newly introduced property to this example?

File: component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java [hunk 2]
```diff
@@ -136,12 +145,22 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
...
         this.optionHolder = optionHolder;
         this.connectionFactory = new MQQueueConnectionFactory();
+        this.outputStreamDefinition = outputStreamDefinition;
         this.queueName = optionHolder.validateAndGetStaticValue(IBMMQConstants.DESTINATION_NAME);
         this.userName = optionHolder.validateAndGetStaticValue(IBMMQConstants.USER_NAME,
                 sinkConfigReader.readConfig(IBMMQConstants.USER_NAME, null));
         this.password = optionHolder.validateAndGetStaticValue(IBMMQConstants.PASSWORD,
                 sinkConfigReader.readConfig(IBMMQConstants.PASSWORD, null));
-
+        this.properties = optionHolder.validateAndGetStaticValue(IBMMQConstants.PROPERTIES, sinkConfigReader.readConfig
```
Comment (pcnfernando) [inline component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java:157]: Can you please check the line length

File: component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java [hunk 3]
```diff
@@ -136,12 +145,22 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
...
+                (IBMMQConstants.PROPERTIES, null));
+        if (properties != null) {
+            try {
+                connectionFactory.setBatchProperties(generatePropertyMap(properties));
+            } catch (JMSException e) {
+                throw new IBMMQSinkAdaptorRuntimeException("Error occurred while initializing IBM MQ sink properties " +
```
Comment (pcnfernando) [inline component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java]: Shall we have a more descriptive exception saying while using properties element in sink configurations.

File: component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java [hunk 4]
```diff
@@ -87,6 +88,13 @@
                         type = DataType.STRING,
                         optional = true,
                         defaultValue = "null"),
+                @Parameter(name = IBMMQConstants.PROPERTIES,
+                        description = "IBM MQ properties can be provided as key value pairs which is separated" +
+                                " by \",\". as an example " +
+                                "properties = 'XMSC_WMQ_CLIENT_RECONNECT_OPTIONS:1600,WMQ_CLIENT_RECONNECT:5005' ",
```
Comment (pcnfernando) [inline component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java]: Shall we point to the official IBMMQ client doc for valid properties. And also let's make a note, these properties only will be applied if the client supports it

**Review by pcnfernando — CHANGES_REQUESTED** [review]
Please address the commented

**Review by minudika — CHANGES_REQUESTED** [review]

**Review by pcnfernando — APPROVED** [review]

File: component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java (hunk 1 above)
Comment (AnuGayan) [reply component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java:113]: FIxed by 671a8172e76badc2d5b132421f3c616fcc58dd60

**Review by minudika — APPROVED** [review]

**Pull request merged**